tools/warpy/post_processing/post2d.py

```diff
@@ -142,6 +142,55 @@ def tripcolor(self, frame, y_fun, **kwargs):
                 # TODO: is m_id actually an integer or not? this is a hack to sidestep the issue
                 raise RuntimeError('invalid mesh id %s found' % (str(m_id)))
 
+    def tricontour(self, frame, y_fun, nlevels=10, levels=None, colors=None, **kwargs):
+        '''
+        @brief Plots contours of y data on triangle grid. Plot style parameters may be passed by name.
+        @param frame integer of what frame to plot
+        @param y_fun What function to call for y values
+        @param subdomains (named only) List of subdomain(s) to extract data for, default is to plot for all subdomains
+        '''
+        dset = self.dset
+        fields = dset.fields[frame]
+        t = fields[0].time
+        sorder = self.sorder
+        shading = 'flat' if sorder == 1 else 'gouraud'
+        if (colors is None):
+            colors = 'k'
+
+        if ('subdomains' in kwargs):
+            subdomains = kwargs['subdomains']
+            subdomains = self.all_subdomains if subdomains is None else subdomains
+            del kwargs['subdomains']
```

Comment:
It would be more readable to simply declare `subdomains` as a kwarg, rather than having it implicit in the splatted `**kwargs` parameter.
```suggestion
def tricontour(self, frame, y_fun, nlevels=10, levels=None, colors='k', subdomains=None, **kwargs):
...
    if subdomains is None:
        subdomains = self.all_subdomains
```